### DISCUSSION TITLE
Alternate driver skins (rivals?)

### DISCUSSION THREAD

**Discussion by Tenome** [emoose/OutRun2006Tweaks#262] [Suggestions]
### Describe the feature you'd like to see added

According to Howard's notes from OutrunX, obj_chr_aut contains alternate costumes for the driver that might be from rivals or from multiplayer. Would it be possible to add these as a selection/random choice?

On a related note, FXT also let you drive as Clarissa and Flag Man, but they were janky from what I remember.

> Chr folder:
> ------------
> Character models and textures
> Note:
> - gal/g00=clarissa 
> - gal_usa/g00_usa=censored clarissa
> - h00=holly
> - fal/l00=jennifer
> - mal/m00=alberto
> - s00=sam (blonde male)
> - w00=wolf (black haired male)
> 
> obj_chr_dr_???_pmt - driver model and textures (use notes above)
> obj_chr_dr_?h??_pmt - hi res driver model and textures 
> obj_chr_???_pmt - models used in endings (unlike the driver versions these have legs and bigger textures)
> 
> obj_chr_aut??_pmt - alternate costumes for the drivers?? used by rivals maybe??
> obj_chr_aut04_pmt - flag man
> obj_chr_auto04_cvt_pmt - ditto
> 
> Something of note is the fact that the driver models have pallette swap textures in their archives changing the hair and shirt colors

> Driver folder:
> --------------
> contains the models and textures for drivers it's unknown if these or those found in the chr folder are used in various in-game modes
> obj_driver_gal_pmt/obj_driver_galusa_pmt - clarissa
> obj_driver_rival_pmt - textures and alternate textures for alberto, clarissa and jennifer
> obj_driver_special_pmt - textures for clarissa, jennifer, sam and wolf

**Comment by espiox** [comment]
At one point in development they planned to let you choose your driver (from the three guys and three girls) and I believe there’s unfinished files for a character select on the license screen?

**Comment by emoose** [comment]
The addr I mentioned in free cam thread can let you switch between a couple chars, think it gets written into savegame as well: `7C2400 (4 bytes)`

IIRC there were some pretty weird ones there as well, maybe some sort of test characters. I wasn't able to get flag man to work though, wonder how FXT did it.

Wasn't able to get the girl chars to work right neither for some reason, don't they show up as rivals? guess maybe they aren't coded to work in player car properly.

Wonder how far they got with character selection, maybe there's some leftovers for it in the code...

E: huh seems kinda usable:

![image](https://github.com/user-attachments/assets/8df7c83a-8640-48a7-b128-86bc3792b162)

Selecting male chars seems to work but girls cause crash when entering race, too bad... Do the later PS2/online arcade ports have anything for this?

**Comment by emoose** [comment]
Seem to have the character choice unlocked properly now, instead of forcing it to the screen like in last post:

![image](https://github.com/user-attachments/assets/9a7a3e1e-ec88-4982-a067-043bc457e232)

If anyone wants to try here's a build: 
[OutRun2006Tweaks-CharaSelectionUnlocker.zip](https://github.com/user-attachments/files/16460966/OutRun2006Tweaks-CharaSelectionUnlocker.zip)

Maybe there's a chance the girls can work on certain stages/modes, otherwise not really sure if this would be worth adding to Tweaks...

**Comment by envido32** [comment]
> If anyone wants to try here's a build: [OutRun2006Tweaks-CharaSelectionUnlocker.zip](https://github.com/user-attachments/files/16460966/OutRun2006Tweaks-CharaSelectionUnlocker.zip)
> 
> Maybe there's a chance the girls can work on certain stages/modes, otherwise not really sure if this would be worth adding to Tweaks...

I haven't been able to load this built. I tried with a clean installation and with admin privileges. v0.5a and v0.4.2 works just fine....

![image](https://github.com/user-attachments/assets/ebc57555-2ab7-414b-b2a2-7f6d63d49daa)

But I did spend some time trying the models by editing the SaveFile: Wolf seems to work just fine, Sam has a small animation issue on his legs... not a big deal but noticeable.

The girls as drivers on the other hand has their animations all broken. It doesn't crash but it looks awful... but they seem to have the same kind of "error", so my speculation is that fix one would fix all of the girls... if possible.

I will keep looking to see if the animations can be swaped or something... maybe some extra testing on different game modes and how it affects online for example.

**Comment by emoose** [comment]
Hm it still seems to run fine for me, that error would usually mean an issue with setting up the patches/hooks themselves, do you get any outrun2006tweaks.log file from it? Also are you using the OR2006C2C.exe from the tweaks DL?

Now we have online working maybe should look into adding this properly, tried seeing if I could hide the broken girl selections but no luck yet...
(Really it's kinda weird girls are broken, don't they show up as rivals and have driving anims etc?)

**Comment by emoose** [comment]
Ah there were a couple extra changes in that build which could have broken something, here's a new build with just the license screen stuff:
[OutRun2006Tweaks-CharaSelectionUnlockerv2.zip](https://github.com/user-attachments/files/16563733/OutRun2006Tweaks-CharaSelectionUnlockerv2.zip)

**Comment by envido32** [comment]
> Ah there were a couple extra changes in that build which could have broken something, here's a new build with just the license screen stuff: [OutRun2006Tweaks-CharaSelectionUnlockerv2.zip](https://github.com/user-attachments/files/16563733/OutRun2006Tweaks-CharaSelectionUnlockerv2.zip)

Weird... same error.
I uploaded the log file if you want to take a look

[OutRun2006Tweaks.log](https://github.com/user-attachments/files/16570958/OutRun2006Tweaks.log)

**Comment by emoose** [comment]
Hm, seems it's not even getting to apply the hooks, weird.

Just noticed there was some debug logging stuff in that code too, maybe that can cause issues on some systems.. 

Maybe see if this is any better: 
[OutRun2006Tweaks-CharaSelectionUnlockerv3.zip](https://github.com/user-attachments/files/16571063/OutRun2006Tweaks-CharaSelectionUnlockerv3.zip)

**Comment by Tenome** [comment]
I can't test it because it has that other issue that you already fixed, but it looks like it still works. Can you implement it in the main branch? I brought the driver selection screen but then didn't notice that it wasn't actually made an official feature yet lol.

![image](https://github.com/user-attachments/assets/243fc2f5-6a3d-4c00-bce9-4899a7be3526)

**Comment by Tenome** [comment]
> I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.

Yes, since that's just a texture replacement. The models aren't ripped though so you'll have to do that yourself (NinjaRipper might work), otherwise you can just color-correct the hair/shirt and see how it looks in-game.

**Comment by Tenome** [comment]
> I wonder if it'll be possible to change the colors of default in-game models, since TCRF mentions leftovers of a redheaded Jennifer and a brunette Clarissa.

Swap the filenames of the textures and boom, redhead.

![OR2006C2C_n3WGdvfnsV](https://github.com/user-attachments/assets/a83414ed-d5bd-4165-b968-725779aa2868)

**Comment by Tenome** [comment]
Do you mean brunette Jennifer? Yep.
![OR2006C2C_FDyuSTQeGV](https://github.com/user-attachments/assets/ca8898dc-5d13-4c48-9cdf-2dad74c02dec)

**Comment by envido32** [comment]
Wow! This is really interesting!

But if the files are not dumped, where did you ripped from? Are they insisde the game like JP's Clarrissa? In that case... could it be possible to have a toggle in the .ini file to select "Alternative GF"?

**Comment by Tenome** [comment]
The files were dumped via Tweaks, so they're in the game. They're just not used. They're in the same folder as the regular textures.

**Comment by Tenome** [comment]
The files for all driver/passenger skins are in obj_driver_rival_pmt. Swapping them out is simple, but it'd be cool to have it as a Tweaks option too.
